crates/vault-registry/src/tests.rs

```diff
@@ -1353,13 +1353,121 @@ mod get_vaults_with_issuable_tokens_tests {
             // update the exchange rate
             ext::oracle::backing_to_issuing::<Test>.mock_safe(move |x| MockResult::Return(Ok(x / 2)));
 
-            assert_err!(
-                VaultRegistry::get_vaults_with_issuable_tokens(),
-                TestError::NoVaultWithIssuableTokens
+            assert_eq!(VaultRegistry::get_vaults_with_issuable_tokens(), Ok(vec!()));
+        })
+    }
+}
+
+mod get_vaults_with_redeemable_tokens_test {
+    use super::*;
+
+    #[test]
+    fn get_vaults_with_redeemable_tokens_returns_empty() {
+        run_test(|| {
+            create_sample_vault();
+            // nothing issued, so nothing can be redeemed
+            assert_eq!(VaultRegistry::get_vaults_with_redeemable_tokens(), Ok(vec!()));
+        })
+    }
+
+    #[test]
+    fn get_vaults_with_redeemable_tokens_succeeds() {
+        run_test(|| {
+            let id1 = 3;
+            create_vault(id1);
+            let issued_tokens1: u128 = 10;
+            VaultRegistry::try_increase_to_be_issued_tokens(&id1, issued_tokens1).unwrap();
+            assert_ok!(VaultRegistry::issue_tokens(&id1, issued_tokens1));
+            let vault1 = VaultRegistry::get_active_rich_vault_from_id(&id1).unwrap();
+            assert_eq!(vault1.data.issued_tokens, issued_tokens1);
+            assert_eq!(vault1.data.to_be_redeemed_tokens, 0);
```

Comment:
Great to see these tests! They're a bit verbose though, maybe it would be nice to extract the 8 lines above into a reusable function inside of the get_vaults_with_redeemable_tokens_test module?